src/shogun/base/SGObject.h

```diff
@@ -390,9 +403,38 @@ class CSGObject
 		put(Tag<T*>(name), value);
 	}
 
+	/** Typed setter for an enum mapped with a string, identified by a name.
+	 *
+	 * @param name name of the parameter
+	 * @param value value of the parameter
+	 */
+	template <class T>
+	void
+	put(const std::string& name, const std::string& value) noexcept(false)
+	{
+		if (m_string_to_enum_map.find(name) == m_string_to_enum_map.end())
+		{
+			SG_ERROR(
+				"There are no options for the parameter %s::%s", get_name(),
+				name.c_str());
+		}
+
+		auto string_to_enum = m_string_to_enum_map[name];
+
+		if (string_to_enum.find(value) == string_to_enum.end())
+		{
+			SG_ERROR(
+				"There is no enum mapping for '%s' in %s::%s",
```

Comment:
could you rather say something like "illegal option %s for parameter %s::%s" ... ?

Reply:
OK! Was just trying to follow the pattern from the other error messages